controllers/machinehealthcheck_controller_test.go

```diff
@@ -111,841 +110,1004 @@ func newInfraMachine(machine *clusterv1.Machine) (*unstructured.Unstructured, st
+		}(cluster, mhc, infraTmpl, machineSet)
+
+		// Pause the MachineSet reconciler. WHY?? I SEE WHAT IS HAPPENING BUT
+		// WHY DO WE PAUSE THIS?
+		machineSetPatch := client.MergeFrom(machineSet.DeepCopy())
+		machineSet.Annotations = map[string]string{
+			clusterv1.PausedAnnotation: "",
+		}
+		g.Expect(testEnv1.Patch(ctx, machineSet, machineSetPatch)).To(Succeed())
+
+		// Calculate how many Machines should be remediated.
+		var unhealthyMachine *clusterv1.Machine
+		g.Eventually(func() (remediated int) {
+			machines := &clusterv1.MachineList{}
+			err := testEnv1.List(ctx, machines, client.MatchingLabels{
+				"selector": mhc.Spec.Selector.MatchLabels["selector"],
+			})
+			if err != nil {
+				return -1
 			}
-			Expect(testEnv.Patch(ctx, machineSet, machineSetPatch)).To(Succeed())
-
-			// Calculate how many Machines should be remediated.
-			var unhealthyMachine *clusterv1.Machine
-			Eventually(func() (remediated int) {
-				machines := &clusterv1.MachineList{}
-				err := testEnv.List(ctx, machines, client.MatchingLabels{
-					"selector": mhc.Spec.Selector.MatchLabels["selector"],
-				})
-				if err != nil {
-					return -1
-				}
 
-				for i := range machines.Items {
-					if conditions.Get(&machines.Items[i], clusterv1.MachineOwnerRemediatedCondition) != nil {
-						unhealthyMachine = machines.Items[i].DeepCopy()
-						remediated++
-					}
+			for i := range machines.Items {
+				if conditions.Get(&machines.Items[i], clusterv1.MachineOwnerRemediatedCondition) != nil {
+					unhealthyMachine = machines.Items[i].DeepCopy()
+					remediated++
 				}
-				return
-			}).Should(Equal(1))
-
-			// Unpause the MachineSet reconciler.
-			machineSetPatch = client.MergeFrom(machineSet.DeepCopy())
-			delete(machineSet.Annotations, clusterv1.PausedAnnotation)
-			Expect(testEnv.Patch(ctx, machineSet, machineSetPatch)).To(Succeed())
-
-			// Make sure the Machine gets deleted.
-			Eventually(func() bool {
-				machine := unhealthyMachine.DeepCopy()
-				err := testEnv.Get(ctx, util.ObjectKey(unhealthyMachine), machine)
-				return apierrors.IsNotFound(err) || !machine.DeletionTimestamp.IsZero()
-			})
-		})
+			}
+			return
+		}, timeout, time.Second).Should(Equal(1))
+
+		// Unpause the MachineSet reconciler.
+		machineSetPatch = client.MergeFrom(machineSet.DeepCopy())
+		delete(machineSet.Annotations, clusterv1.PausedAnnotation)
+		g.Expect(testEnv1.Patch(ctx, machineSet, machineSetPatch)).To(Succeed())
+
+		// Make sure the Machine gets deleted.
+		g.Eventually(func() bool {
+			machine := unhealthyMachine.DeepCopy()
+			err := testEnv1.Get(ctx, util.ObjectKey(unhealthyMachine), machine)
+			return apierrors.IsNotFound(err) || !machine.DeletionTimestamp.IsZero()
+		}, timeout, time.Second)
+	})
 
-		Specify("when a machine is paused", func() {
-			mhc.Spec.ClusterName = cluster.Name
-			Expect(testEnv.Create(ctx, mhc)).To(Succeed())
+	t.Run("when a machine is paused", func(t *testing.T) {
+		t.Skip("skipping until made stable")
```

Comment:
There are a few skipped tests. These tests are very unstable. Will review them after some other conditions related work gets merged in.